src/completions.rs

```diff
@@ -0,0 +1,224 @@
+use anyhow::{anyhow, Result};
+use log::{error, info, warn};
+use schemars::{schema_for, JsonSchema};
+use serde::{de::DeserializeOwned, Serialize};
+use serde_json::Value;
+
+use crate::llm_models::LLMModel;
+use crate::{domain::OpenAIDataResponse, utils::get_tokenizer};
+
+/// Completions APIs take a list of messages as input and return a model-generated message as output.
+/// Although the Completions format is designed to make multi-turn conversations easy,
+/// it’s just as useful for single-turn tasks without any conversation.
+pub struct Completions<T: LLMModel> {
```

Comment:
Most of this "new" module is really not new. It's a direct copy of previously existing `openai` module. The only changes that were made were to make it accept any enum that implements the new `LLMModel` trait.